cas-server-core/src/main/java/org/jasig/cas/AbstractCentralAuthenticationService.java

```diff
@@ -150,6 +159,15 @@ protected final void doPublishEvent(final ApplicationEvent e) {
 
 
     /**
+     * Gets a synchronization handle.
+     * @param id String mutex id
+     * @return SimpleMutex
+     */
+    protected final SimpleMutex getMutex(final String id) {
```

Comment:
you are right about the weakness of the objects. a `WeakHashMap` only stores weak links to the key. so, lifetime of the mutex object would be tied to the `String` key.